File: backend/src/main/java/com/decmoe47/todo/service/AuthService.java
package com.decmoe47.todo.service;

import com.decmoe47.todo.model.dto.UserLoginDTO;
import com.decmoe47.todo.model.dto.UserRegisterDTO;
import com.decmoe47.todo.model.vo.AuthenticationTokensVO;
import com.decmoe47.todo.model.vo.UserVO;

public interface AuthService {

    UserVO login(UserLoginDTO userLoginDTO);

    void logout(String token);

    UserVO register(UserRegisterDTO userRegisterDTO);

    void sendVerificationCode(String email);

    AuthenticationTokensVO refreshAccessToken(String refreshToken);
}

File: backend/src/main/kotlin/com/decmoe47/todo/service/AuthService.kt
package com.decmoe47.todo.service

import com.decmoe47.todo.model.request.UserLoginRequest
import com.decmoe47.todo.model.request.UserRegisterRequest
import com.decmoe47.todo.model.response.AuthenticationTokensResponse
import com.decmoe47.todo.model.response.UserResponse

interface AuthService {
    fun login(request: UserLoginRequest): UserResponse

    fun logout(token: String)

    fun register(request: UserRegisterRequest): UserResponse

    fun sendVerificationCode(email: String)

    fun refreshAccessToken(refreshToken: String): AuthenticationTokensResponse
}